### PR TITLE
Ui improvements

### PR DESCRIPTION
**Pull request by janno42**
A couple of random UI improvements. See commit messages for details.

### CODE REVIEW

File: evap/results/templates/results_index_course.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
                 <span data-toggle="tooltip" data-placement="top" class="fas fa-chart-bar icon-yellow" title="{% trans 'Results not yet published' %}"></span>
             {% endif %}
         {% endif %}
-        <span class="course-name">{{ course.name }}</span>
+        <span class="course-name">{{ course.name }}{% if course.is_single_result %} ({{ course.vote_start_datetime|date:"Y-m-d" }}){% endif %}</span>
```
Comment (karyon) [inline evap/results/templates/results_index_course.html]: you should be able to leave out the parameter to `date`, the default is `DATE_FORMAT`, which is exactly what you wrote there.

File: evap/evaluation/templates/base.html [hunk 2]
```diff
@@ -97,13 +97,17 @@
...
             language: "{{ LANGUAGE_CODE }}",
             placeholder: "{% trans 'Please select...' %}",
-            minimumResultsForSearch: 15,});
+            minimumResultsForSearch: 15,
+            minimumInputLength: minimumInput ? 3 : 0
+        });
     };
-    applySelect2($('select').not('.form-template select').not('[disabled]'));
+    $('select').not('.form-template select').not('[disabled]').each(function() {
+        applySelect2($(this), $(this).children('option').length > 50);
```
Comment (karyon) [inline evap/evaluation/templates/base.html]: since the parameter is set only here, i would inline that.

File: evap/evaluation/tools.py [hunk 3]
```diff
@@ -134,3 +134,13 @@ def get_due_courses_for_user(user):
...
+
+
+def get_parameter_from_url_or_session(request, parameter):
+    result = request.GET.get(parameter, None)
+    if result is None:  # if no parameter is given take session value
+        result = request.session.get(parameter, False)  # defaults to False if no session value exists
+    else:
+        result = {'true': True, 'false': False}.get(result.lower())  # convert parameter to boolean
+    request.session[parameter] = result  # store value for session
+    return result
```
Comment (karyon) [inline evap/evaluation/tools.py]: you can squash that into the next commit, it's a really trivial refactoring.

Comment (karyon) [reply evap/evaluation/tools.py]: ...disregard that, i missed the changes in the next commit.

**Comment by janno42** [comment]
About hiding delegated courses: Some users (like me) have a huge list of courses on the contributor index page because they are set as delegates. That is perfectly acceptable when preparing the evaluation for all these courses. But when I'm just looking for my own evaluation results, a filtered list is quite nice because I can then get a fast overview and much faster find which courses are really my "Own courses".

We could also introduce a "delegation mode" similar to what we're planning for staff, but that's probably way too much overhead.

File: evap/results/templates/results_index_course.html (hunk 1 above)
Comment (janno42) [reply evap/results/templates/results_index_course.html]: leaving that out returns `yyyy-MM-dd HH:mm`

File: evap/evaluation/templates/base.html (hunk 2 above)
Comment (janno42) [reply evap/evaluation/templates/base.html]: like how?

Comment (janno42) [reply evap/evaluation/templates/base.html]: ah, i think i know what you mean.

Comment (karyon) [reply evap/evaluation/templates/base.html]: i'm not quite sure what you're asking, i was thinking something like `minimumInputLength: elem..children('option').length > 50 ? 3 : 0`

actually, if i understand correctly, without that you'd be introducing a bug. `applySelect2` is called e.g. when adding a form to the contribution formset on the course edit page, and the added forms would have a `minimumInputLength` of 0 while the initially visible forms would have set that to 3.

File: evap/results/templates/results_index_course.html (hunk 1 above)
Comment (karyon) [reply evap/results/templates/results_index_course.html]: that is weird. https://docs.djangoproject.com/en/2.1/ref/templates/builtins/#date says `When used without a format string, the DATE_FORMAT format specifier is used`, and from a quick glance at the code, it does just that. could you investigate?

Comment (janno42) [reply evap/results/templates/results_index_course.html]: it's not a date, but a datetime ;)

Comment (karyon) [reply evap/results/templates/results_index_course.html]: yes, but the `date` filter does not care as far as i can see.

File: evap/evaluation/templates/base.html (hunk 2 above)
Comment (karyon) [reply evap/evaluation/templates/base.html]: does that still work when calling `applySelect2` with a single element like when adding a form?

Comment (janno42) [reply evap/evaluation/templates/base.html]: yes. e.g., on the course edit page i can add additional contributors and the selection field there still requires three characters to be typed before showing options.

File: evap/contributor/views.py [hunk 4]
```diff
@@ -17,17 +17,20 @@
...
     own_courses = Course.objects.filter(contributions__contributor=user, state__in=contributor_visible_states)
 
     represented_users = user.represented_users.all()
     delegated_courses = Course.objects.exclude(id__in=own_courses).filter(contributions__can_edit=True, contributions__contributor__in=represented_users, state__in=contributor_visible_states)
 
-    all_courses = list(own_courses) + list(delegated_courses)
-    all_courses.sort(key=lambda course: list(STATES_ORDERED.keys()).index(course.state))
+    displayed_courses = list(own_courses)
+    if show_delegated:
+        displayed_courses += list(delegated_courses)
```
Comment (karyon) [inline evap/contributor/views.py]: right now, if the user chose to not show delegated courses, you are still getting them only to check in the template whether any such course exists.
i'd move the `delegated_courses` here and check whether the user `is_delegate` in the template.

Comment (janno42) [reply evap/contributor/views.py]: done and i completely removed the delegated courses from the template data by adding an attribute to the delegated courses.

**Pull request merged**